libpathod/protocols/http.py

```diff
@@ -70,4 +68,8 @@ def handle_http_connect(self, connect, lg):
         return self.pathod_handler.handle_http_request, None
 
     def read_request(self, lg=None):
-        return self.wire_protocol.read_request(allow_empty=True)
+        try:
+            return http1.read_request(self.pathod_handler.rfile)
+        except HttpReadDisconnect:
+            # TODO: This is
+            return Request("", b"", b"", b"", b"", b"", b"", None, b"")
```

Comment:
Fixed - we now let the HttpReadDisconnect error  go through and act here accordingly.